db/db_impl/db_impl.cc

```diff
@@ -5236,7 +5236,23 @@ Snapshot::~Snapshot() = default;
 
 Status DestroyDB(const std::string& dbname, const Options& options,
                  const std::vector<ColumnFamilyDescriptor>& column_families) {
-  ImmutableDBOptions soptions(SanitizeOptions(dbname, options));
+  Options options_copy = options;
+  auto sfm =
+      static_cast<SstFileManagerImpl*>(options_copy.sst_file_manager.get());
+  if (sfm) {
+    // Use a new SstFileManager with the same configuration if there is an
+    // existing SstFileManager since DestroyDB need to independently track and
```

Comment:
Why does it need to be independent? Wouldn't it be better to just slow delete under the same manager, so that we don't double-up deletion rate on a single device?

Reply:
Yeah, that's a good point, the user could have configured the original `SstFileManager` to be shared. This was intended to make the `DestroyDB`'s`SstFileManager` to do its independent accounting. I think that's not necessary then given this side effect.